### PR TITLE
fix: Rounding issues in score_answer and add unit tests

### PR DESCRIPTION
**Pull request by Adefioye**
This helps to address the issue raised [here](https://github.com/open-thought/reasoning-gym/issues/428).

### CODE REVIEW

File: reasoning_gym/arithmetic/power_function.py [hunk 1]
```diff
@@ -37,26 +37,35 @@ class PowerFunctionDataset(ProceduralDataset):
...
-                oracle_value = Decimal(oracle_answer)
-
-                if oracle_value == 0:
-                    return 1.0 if user_answer == 0 else 0.01
+                user_answer = self._format_sig_figs(Decimal(answer), 3)
+                oracle_answer = self._format_sig_figs(Decimal(oracle_answer), 3)
 
-                user_sig_figs = f"{user_answer:.3g}"
-                oracle_sig_figs = f"{oracle_value:.3g}"
+                print(f"user answer: {user_answer}, oracle answer: {oracle_answer}")
```
Comment (olliestanley) [inline reasoning_gym/arithmetic/power_function.py:60]: We should remove these debug prints before merging

File: tests/test_power_function.py [hunk 2]
```diff
@@ -82,3 +84,19 @@ def test_power_function_curriculum():
...
+    dataset = PowerFunctionDataset(config)
+    item = dataset[0]  # Get the first item for testing
+    assert isinstance(item, dict)
+
+    user_answer = "1.000e+00"
+    # Let's change the oracle answer for edge case testing
+    item["answer"] = "1.0"
+    score = dataset.score_answer(user_answer, item)
+    print(f"Score for edge case answer: {score}")
+    assert score == 1.0, f"Expected score 1.0, got {score}"
```
Comment (olliestanley) [inline tests/test_power_function.py:102]: Maybe we could add 2/3 extra cases to this test?

File: reasoning_gym/arithmetic/power_function.py (hunk 1 above)
Comment (Adefioye) [reply reasoning_gym/arithmetic/power_function.py:60]: Woops! My bad will remove it.

**Review by olliestanley — APPROVED** [review]
